### PR TITLE
Set type as both component or endpoint type, as will be relevant for …

### PR DESCRIPTION
**Pull request by objectiser**
…either

### CODE REVIEW

File: client/opentracing/src/main/java/org/hawkular/apm/client/opentracing/DefaultNodeProcessor.java [hunk 1]
```diff
@@ -45,7 +45,9 @@ public void process(TraceContext context, APMSpan span, NodeBuilder nodeBuilder)
                     } catch (MalformedURLException e) {
                         nodeBuilder.setUri(entry.getValue().toString());
                     }
-                    nodeBuilder.setEndpointType(entry.getKey().substring(0, entry.getKey().length() - 4));
+                    String type=entry.getKey().substring(0, entry.getKey().length() - 4);
```
Comment (lucasponce) [inline client/opentracing/src/main/java/org/hawkular/apm/client/opentracing/DefaultNodeProcessor.java]: perhaps a if (entry.getKey.length() >=4) validation somewhere ?
But just if it makes sense.

Comment (objectiser) [inline client/opentracing/src/main/java/org/hawkular/apm/client/opentracing/DefaultNodeProcessor.java]: there is already a check for key ending ".url" or ".uri" - so would always be atleast 4

**Comment by lucasponce** [comment]
LGTM

**Comment by objectiser** [comment]
@lucasponce Thanks!

**Pull request merged**